pkg/controller/daemon/util/daemonset_util.go

```diff
@@ -199,3 +199,51 @@ func ReplaceDaemonSetPodHostnameNodeAffinity(affinity *v1.Affinity, nodename str
 
 	return affinity
 }
+
+// AppendNoScheduleTolerationIfNotExist appends unscheduable toleration to `.spec` if not exist; otherwise,
+// no changes to `.spec.tolerations`.
+func AppendNoScheduleTolerationIfNotExist(tolerations []v1.Toleration) []v1.Toleration {
+	unschedulableToleration := v1.Toleration{
+		Key:      algorithm.TaintNodeUnschedulable,
+		Operator: v1.TolerationOpExists,
+		Effect:   v1.TaintEffectNoSchedule,
+	}
+
+	unschedulableTaintExist := false
+
+	for _, t := range tolerations {
```

Comment:
What if `t` is a unschedulable toleration whose `Operator` is `Equal` instead of `Exists`?

Reply:
if so, we are going to append unschedulable toleration.